### ISSUE TITLE
Bug: Validation:validateRequired() get an error when field name is an array

### ISSUE THREAD

**Issue by altsniffer** [mako-framework/framework#231]
Hi

I get an error when i'm trying to use validation required on a checkbox "name='foo[]'", is that wanted to be a string?

regards

**Comment by freost** [comment]
I'm planning to rewrite the validation library at some point (hopefully soon) to add support for array validation.

**Comment by freost** [comment]
5.5 will include improved validation with support for arrays.

You can now also validate values inside nested arrays using "dot notation":

```php
$rules = 
[
    'foo.bar' => ['required']
];
```

I'll also be adding rules to validate `SplFileInfo` objects so that it will be possible to validate uploaded files as well.

The only backwards compatibility break is how you register custom validation rules:

```php
$this->validator->extend('myrule', MyRule::class);
```

I haven't committed the unit tests yet but everything will have tests before being released.

**Comment by freost** [comment]
A new and unit tested validation library with array support is available in the 5.5 branch (master).
